chain/client/src/view_client.rs

```diff
@@ -246,6 +250,62 @@ impl ViewClientActor {
         }
     }
 
+    /// Returns maintenance windows by account.
+    fn get_maintenance_windows(
+        &self,
+        account_id: AccountId,
+        genesis_config: GenesisConfig,
+    ) -> Result<MaintenanceWindowsView, near_chain::Error> {
+        let epoch_manager = EpochManager::new_from_genesis_config(
+            self.runtime_adapter.store().clone(),
+            &genesis_config,
+        )?;
+        let chain_store = ChainStore::new(
+            self.runtime_adapter.store().clone(),
+            genesis_config.genesis_height,
+            false,
+        );
+
+        let head = chain_store.head()?;
+        let epoch_id = epoch_manager.get_epoch_id(&head.last_block_hash)?;
+        let epoch_info: Arc<EpochInfo> = epoch_manager.get_epoch_info(&epoch_id)?;
+        let num_shards = self.runtime_adapter.num_shards(&epoch_id)?;
+        let cur_block_info = epoch_manager.get_block_info(&head.last_block_hash)?;
+        let next_epoch_start_height = epoch_manager
+            .get_epoch_start_height(cur_block_info.hash())?
+            + epoch_manager.get_epoch_config(&epoch_id)?.epoch_length;
+
+        let mut windows: MaintenanceWindowsView = Vec::new();
+        let mut start_block_of_window: Option<BlockHeight> = None;
+        let last_block_of_epoch = next_epoch_start_height - 1;
+
+        for block_height in head.height..next_epoch_start_height {
```

Comment:
I'd mention in the RPC & this function description, that you're looking for windows from the current head till the end of the epoch.

Reply:
Yes, exactly.  

The past windows are not useful for maintenance. I intend to remove these windows.  Also, I mentioned the *future* maintenance window in the doc.
https://github.com/near/docs/pull/1277/files#diff-6ed641db4bb40f6ce8ab6fd360504141e710d6529f538d04adf7b1b4eddfd6f8R9

Such that the user using this RPC will not need an additional query for the current block height.

If there is any concern, we can change is to all the windows in the current epoch.